common/djangoapps/third_party_auth/tests/test_provider.py

```diff
@@ -84,6 +90,41 @@ def test_providers_displayed_for_login(self):
         self.assertNotIn(no_log_in_provider.provider_id, provider_ids)
         self.assertIn(normal_provider.provider_id, provider_ids)
 
+    def test_providers_enabled_for_site(self):
+        """
+        Test the logic of the enabled_for_current_site property to ensure that a provider can only be
+        used when making requests in the context of its corresponding site.
+        """
+        site_a, __ = Site.objects.get_or_create(domain=SITE_DOMAIN_A, name=SITE_DOMAIN_A)
+        site_b, __ = Site.objects.get_or_create(domain=SITE_DOMAIN_B, name=SITE_DOMAIN_B)
+
+        default_site_provider = self.configure_google_provider(visible=True, enabled=True)
+        default_site_disabled_provider = self.configure_google_provider(visible=True, enabled=False)
+        site_provider = self.configure_google_provider(visible=True, enabled=True, site=site_a)
+        site_disabled_provider = self.configure_google_provider(visible=True, enabled=False, site=site_a)
```

Comment:
@bdero `configure_google_provider` creates an `OAuth2ProviderConfig` and if I recall right `OAuth2ProviderConfig` is a versioned model (inherited from `ConfigurationModel`), so any updates/insertes to the entity with same "key" (site and provider name in this case) from app point of view update the "current" record. So, you might not need to do it twice: `self.configure_google_provider(visible=True, enabled=False, site=site_a)` should be enough. 

But I might be wrong.